src/covid_model_seiir_pipeline/lib/cli_tools/performance_logger/performance_logger.py

```diff
@@ -0,0 +1,47 @@
+from collections import defaultdict
+import time
+
+from loguru import logger
+
+
+class TaskPerformanceLogger:
+
+    def __init__(self):
+        self.current_context = None
+        self.current_context_start = None
+        self.times = defaultdict(float)
+
+    def _record_timing(self, context):
+        if context is None:
+            return
+        if self.current_context is None:
+            self.current_context = context
+            self.current_context_start = time.time()
+        else:
+            self.times[self.current_context] += time.time() - self.current_context_start
+            self.current_context = context
+            self.current_context_start = time.time()
```

Comment:
This logic means you need to be careful not to repeat contexts or you'll lose time, right?

Reply:
I'll only lose the time of the execution of these three lines as times are cumulative.